studies/forms.py

```diff
@@ -46,6 +50,59 @@ class Meta:
 
 PROTOCOL_HELP_TEXT_INITIAL = f"{PROTOCOL_HELP_TEXT_EDIT}  You can leave the default for now and come back to this later."
 
+DEFAULT_GENERATOR = """function generateProtocol(child, pastSessions) {
```

Comment:
Do we want to have tests to see that the default generator always parses (would probably require a dev dependency like [pyjsparser](https://github.com/PiotrDabkowski/pyjsparser)), or is it pretty much not going to change after this? Up to you, just throwing that out there.

Reply:
So my feeling on server-side validation here of the javascript overall was that the extra dependencies we'd need wouldn't quite be worth the benefit, and would introduce a tighter coupling between ember-lookit-frameplayer and lookit-api than we really want to deal with maintaining. Only introducing this for tests is an interesting compromise - but yeah, I think the default pretty much shouldn't change after this. (It also shouldn't be used directly, so it's essentially a cosmetic issue if it's ever flagged as invalid, albeit one we should avoid!) I'll keep it in mind in case we have any more cases where JS parsing would be relevant, or upon changing the default. 

Eventually it'd definitely be helpful on a variety of templates to be able to test the JS that's *on the template* in context - e.g. render the page, do some action, check that the right thing is showing up.

Reply:
sounds reasonable to me!